datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -246,6 +268,11 @@ pub trait JoinHashMapType {
         let next_chain = self.get_list();
         for (row_idx, hash_value) in iter {
             // Get the hash and find it in the index
+            if let Some(bloom_filter) = self.get_bloom_filter() {
+                if !bloom_filter.contains(hash_value) {
```

Comment:
So the main "problem" here I think is not pushing down the filter, `bloom_filter.contains` probably is about as expensive as `hash_map.get`, so only more overhead is created to create / probe the set.

Reply:
it could be adaptive. eg when the filter is observed not to filter out stuff, it could disable itself (for ever, or "for couple batches")

Reply:
Yes it could be adaptive, however what I'm saying is that because it is directly used in hashjoin itself, there is no actual performance benefit.
It needs to be pushed down below repartition / aggregate / scan etc. to be of any benefit (https://github.com/apache/datafusion/pull/13054)

Reply:
> It needs to be pushed down below repartition / aggregate / scan etc. to be of any benefit

That's a good point.

if the dynamic filter was range-based and we could push it down to the file scan, it could allow file- and row-group-level pruning in Parquet.